### ISSUE TITLE
Blockingproxy: TypeError: Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by rodrigc** [angular/protractor#4473]
I have been using some protractor scripts which have been running fine for a few months.

In the past week, I upgraded to:

- Node Version: `v8.4.0`
- Protractor Version: `version 5.1.2`
- Browser(s): `Chrome 61.0.3163.79`
- Chromedriver: 2.32
- selenium standalone server: 3.5.3
- Operating System and Version `MacOS X and Windows`

and suddenly I am getting this error in BlockingProxy:

```
BlockingProxy error: E/BlockingProxy - (node:26020) TypeError: Cannot read property 'toString' of null
```

I added the `node --trace-warnings` flag and got a stack trace:

```
 E/BlockingProxy - (node:26296) TypeError: Cannot read property 'toString' of null
    at replyWithError (/Users/c-craigr/work1/es-automation-test/ui-tests/node_modules/blocking-proxy/built/lib/webdriver_proxy.js:42:35)
    at WebDriverProxy.<anonymous> (/Users/c-craigr/work1/es-automation-test/ui-tests/node_modules/blocking-proxy/built/lib/webdriver_proxy.js:53:17)
    at Generator.throw (<anonymous>)
    at rejected (/Users/c-craigr/work1/es-automation-test/ui-tests/node_modules/blocking-proxy/built/lib/webdriver_proxy.js:5:62)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

I tracked the problem down to this code in BlockingProxy:

https://github.com/angular/blocking-proxy/blob/master/lib/webdriver_proxy.ts#L29

```
 let replyWithError = (err) => {
      response.writeHead(500);
      response.write(err.toString());
      response.end();
    };
```

In this case `err` is null, so that causes an exception to be thrown.

Does anybody have an idea why err is null?

**Comment by rodrigc** [comment]
@adelitofarias I cannot see your full config file.  The problem seems to be in BlockingProxy, which is available in Protractor 5.1.0 and higher.
Disabling BlockingProxy made things work for me.
I put this in my config:

```
useBlockingProxy: false
```
and removed `webDriverLogDir`.

**Comment by rodrigc** [comment]
Moved to this issue: https://github.com/angular/blocking-proxy/issues/38

**Comment by TWiStErRob** [comment]
While angular/blocking-proxy#38 was fixed in `blocking-proxy@1.0.0` and `blocking-proxy@1.0.1`, the currently available released `protractor@5.2.0` version is not using those. To reap the benefits of the fix (using `selenium-webdriver@3.6.x` + logging + highlight) without having to update protractor to the unreleased version here's a **hack** that allows for running tests with blocking proxy (`package.json`):
```json
	"scripts": {
		"postinstall": "rpm run blocking-proxy-hack",
		"postupdate": "rpm run blocking-proxy-hack",
		"blocking-proxy-hack": "rimraf node_modules/protractor/node_modules/blocking-proxy"
	},
	"dependencies": {
		"blocking-proxy": "^1.0.1",
		"protractor": "5.2.0",
		"selenium-webdriver": "^3.6.0"
	},
	"devDependencies": {
		"rimraf": "2.6.2"
	}
```
The way this works is that it installs the new blocking-proxy and it forces the old version to be moved inside `protractor`'s `node_modules`. Using this information we can `rm -rf` that folder to in turn force `protractor` to use the version specified in `dependencies`.

There might be a better way, I'm open to suggestions.
